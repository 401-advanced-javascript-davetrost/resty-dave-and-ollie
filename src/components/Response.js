import React from 'react';
import PropTypes from 'prop-types';

export default function Response({ response }) {
  return (
    <pre>{response}</pre>
  );
}

Response.propTypes = {
  response: PropTypes.string
};
