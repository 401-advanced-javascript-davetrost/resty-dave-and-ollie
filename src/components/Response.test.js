import React from 'react';
import { shallow } from 'enzyme';
import Response from './Response';

describe('Response component', () => {
  it('renders a response', () => {
    const responseExampleString = `[
      {
          "_id": "5da237699734fdcb7bef8f56",
          "name": "Miles Shortman",
          "image": "https://vignette.wikia.nocookie.net/heyarnold/images/2/2e/Miles.jpg/revision/latest/scale-to-width-down/243?cb=20101115140639"
      }
    ]`;
    const wrapper = shallow(<Response response={responseExampleString}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
