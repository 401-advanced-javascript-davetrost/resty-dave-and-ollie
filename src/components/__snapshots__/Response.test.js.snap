// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Response component renders a response 1`] = `
<pre
  styles="Response"
>
  [
      {
          "_id": "5da237699734fdcb7bef8f56",
          "name": "Miles Shortman",
          "image": "https://vignette.wikia.nocookie.net/heyarnold/images/2/2e/Miles.jpg/revision/latest/scale-to-width-down/243?cb=20101115140639"
      }
    ]
</pre>
`;
